core/server/master/src/main/java/alluxio/master/file/InodeSyncStream.java

```diff
@@ -504,18 +537,29 @@ private void syncExistingInodeMetadata(LockedInodePath inodePath)
       }
       persistingLock.get().close();
 
-      UfsStatus cachedStatus = mStatusCache.getStatus(inodePath.getUri());
+      UfsStatus cachedStatus = null;
+      boolean fileNotFound = false;
+      try {
+        cachedStatus = mStatusCache.getStatus(inodePath.getUri());
+      } catch (FileNotFoundException e) {
```

Comment:
side question: when does `getStatus` return null and when does it throw `FileNotFoundException`? it is not clear to me. Also, throwing an exception is considered expensive. If it is expected, can we indicate negative result by return value?